internal/gengapic/gengapic.go

```diff
@@ -66,7 +66,8 @@ func Gen(genReq *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, er
 		if e < 0 {
 			e = len(s)
 		}
-		switch s[:e] {
+		key, val := s[:e], s[e+1:]
```

Comment:
`e` can be set to `len(s)` above. So, this can panic.

https://play.golang.org/p/6V_5jjOBokQ

Reply:
Good point. There is only one plugin option that isn't a `key=value` pair, `sample-only`, so I'll just move that check up and return an error when the format is incorrect. Thanks for catching that.

Reply:
Done.